### ISSUE TITLE
Interpretor + System.iOS 12.16.1.9 fails to build due to --enable-minimal=reflection_emit

### ISSUE THREAD

**Issue by Digi59404** [xamarin/xamarin-macios#6786]
1. Upgrade Xamarin + Xamarin.iOS to the newest platform beta version
2. Use some form of System.Reflection.Emit that used to work with the msbuild --interpreter command.
3. Errors on startup.

### Expected Behavior
When using iOS Interpreter supported code, the app should run as normal and execute normally. I.e. MongoDB.BSON Should work.

### Actual Behavior

Application loads and fails with a 
error: This mono runtime was configured with --enable-minimal=reflection_emit, so System.Reflection.Emit is not supported.

### Environment

```
=== Visual Studio Community 2019 for Mac (Preview) ===

Version 8.3 Preview (8.3 build 1290)
Installation UUID: 1713bec3-f622-45a9-9951-722627ff2464
	GTK+ 2.24.23 (Raleigh theme)
	Xamarin.Mac 5.16.1.0 (d16-3 / 3de2dad6)

	Package version: 604000169

=== Mono Framework MDK ===

Runtime:
	Mono 6.4.0.165 (2019-06/a7919897e8b) (64-bit)
	Package version: 604000169

=== NuGet ===

Version: 5.2.0.6067

=== .NET Core ===

Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	3.0.0-preview6-27804-01
	2.1.11
SDK: /usr/local/share/dotnet/sdk/3.0.100-preview6-012264/Sdks
SDK Versions:
	3.0.100-preview6-012264
	2.1.700
MSBuild SDKs: /Library/Frameworks/Mono.framework/Versions/6.4.0/lib/mono/msbuild/Current/bin/Sdks

=== Xamarin.Profiler ===

Version: 1.6.10
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

=== Updater ===

Version: 11

=== Apple Developer Tools ===

Xcode 10.3 (14492.2)
Build 10G8

=== Xamarin.Mac ===

Version: 5.16.1.9 (Visual Studio Community)
Hash: a12e76f3
Branch: d16-3
Build date: 2019-07-31 19:34:59-0400

=== Xamarin.iOS ===

Version: 12.16.1.9 (Visual Studio Community)
Hash: a12e76f3
Branch: d16-3
Build date: 2019-07-31 19:35:00-0400

=== Xamarin Designer ===

Version: 16.3.0.147
Hash: c301d21d7
Branch: remotes/origin/d16-3
Build date: 2019-08-07 15:23:02 UTC

=== Xamarin.Android ===

Version: 10.0.0.4 (Visual Studio Community)
Commit: xamarin-android/d16-3/a1f7825
Android SDK: /Users/digi/Library/Developer/Xamarin/android-sdk-macosx
	Supported Android versions:
		8.0 (API level 26)
		8.1 (API level 27)

SDK Tools Version: 26.1.1
SDK Platform Tools Version: 28.0.2
SDK Build Tools Version: 28.0.3

Build Information: 
Mono: mono/mono/2019-06@17ac7dcfc74
Java.Interop: xamarin/java.interop/d16-3@5836f58
LibZipSharp: grendello/LibZipSharp/d16-3@71f4a94
LibZip: nih-at/libzip/rel-1-5-1@b95cf3f
ProGuard: xamarin/proguard/master@905836d
SQLite: xamarin/sqlite/3.27.1@8212a2d
Xamarin.Android Tools: xamarin/xamarin-android-tools/d16-3@cb41333

=== Microsoft Mobile OpenJDK ===

Java SDK: /Users/digi/Library/Developer/Xamarin/jdk/microsoft_dist_openjdk_1.8.0.25
1.8.0-25
Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

=== Android SDK Manager ===

Version: 1.4.0.22
Hash: 0bea4f4
Branch: remotes/origin/d16-3~14
Build date: 2019-08-08 19:01:16 UTC

=== Android Device Manager ===

Version: 1.2.0.105
Hash: 7fdb3d0
Branch: remotes/origin/d16-3~1
Build date: 2019-08-08 19:01:36 UTC

=== Xamarin Inspector ===

Version: 1.4.3
Hash: db27525
Branch: 1.4-release
Build date: Mon, 09 Jul 2018 21:20:18 GMT
Client compatibility: 1

=== Build Information ===

Release ID: 803001290
Git revision: 42b8de4ed6237a823e5795bf3d96d76f006c6a00
Build date: 2019-08-12 17:46:41+00
Build branch: release-8.3
Xamarin extensions: 9de44b711b40469c2e98c604b38b0ae33f2024a4

=== Operating System ===

Mac OS X 10.14.6
Darwin 18.7.0 Darwin Kernel Version 18.7.0
    Thu Jun 20 18:42:21 PDT 2019
    root:xnu-4903.270.47~4/RELEASE_X86_64 x86_64


```

### Build Logs

```
2019-08-15 16:01:30.191 Application.iOS[5245:7233256] error: This mono runtime was configured with --enable-minimal=reflection_emit, so System.Reflection.Emit is not supported.
```

**Comment by rolfbjarne** [comment]
This has already been fixed for the next 12.16 preview.

**Comment by arivoir** [comment]
Same problem here. Glad to see it is fixed.
For me the problem is only present when running with "Debug|iPhone" configuration, in other words, when I try to run the app in a real device. When running in the emulator with "Debug|iPhonSimulator" configuration it works fine.
